src/compiler/core_public.rs

```diff
@@ -28,3 +34,26 @@ impl<TItem: Clone> From<&SortedArray<TItem>> for Vec<TItem> {
         sorted_array._vec.clone()
     }
 }
+
+impl<TItem> Deref for SortedArray<TItem> {
+    type Target = [TItem];
```

Comment:
Ah interesting. I guess I've never really seen the non-reference-type slice type anywhere before. That's not something that can actually exist, right? (An "owned" slice?)